src/Aspire.Hosting/AspireStore.cs

```diff
@@ -0,0 +1,234 @@
+    {
+        ArgumentNullException.ThrowIfNullOrWhiteSpace(filename);
+        ArgumentNullException.ThrowIfNullOrWhiteSpace(sourceFilename);
+
+        if (!File.Exists(sourceFilename))
+        {
+            throw new FileNotFoundException("The source file '{0}' does not exist.", sourceFilename);
+        }
+
+        EnsureDirectory();
+
+        // Strip any folder information from the filename.
+        filename = Path.GetFileName(filename);
+
+        // Delete existing file versions with the same name.
+        var allFiles = Directory.EnumerateFiles(_basePath, filename + ".*");
+
+        foreach (var file in allFiles)
+        {
+            try
+            {
+                File.Delete(file);
+            }
+            catch
+            {
+            }
+        }
+
+        var hashStream = File.OpenRead(sourceFilename);
+
+        // Compute the hash of the content.
+        var hash = SHA256.HashData(hashStream);
+
+        hashStream.Dispose();
+
+        var name = Path.GetFileNameWithoutExtension(filename);
+        var ext = Path.GetExtension(filename);
+        var finalFilePath = Path.Combine(_basePath, $"{name}.{Convert.ToHexString(hash)[..12].ToLowerInvariant()}{ext}".ToLowerInvariant());
+
+        if (!File.Exists(finalFilePath))
+        {
+            File.Copy(sourceFilename, finalFilePath, overwrite: true);
+        }
+
+        return finalFilePath;
+    }
+
+    /// <summary>
+    /// Gets a deterministic file path that is a copy of the content from the provided stream.
+    /// The resulting file name will depend on the content of the stream.
+    /// </summary>
+    /// <param name="filename">A file name to base the result on.</param>
+    /// <param name="contentStream">A stream containing the content.</param>
+    /// <returns>A deterministic file path with the same content as the provided stream.</returns>
+    public string GetFileNameWithContent(string filename, Stream contentStream)
+    {
+        ArgumentNullException.ThrowIfNullOrWhiteSpace(filename);
+        ArgumentNullException.ThrowIfNull(contentStream);
+
+        // Create a temporary file to write the content to.
+        var tempFileName = Path.GetTempFileName();
+
+        // Write the content to the temporary file.
+        using (var fileStream = File.OpenWrite(tempFileName))
+        {
+            contentStream.CopyTo(fileStream);
+        }
+
+        var finalFilePath = GetFileNameWithContent(filename, tempFileName);
+
+        try
+        {
+            File.Delete(tempFileName);
+        }
+        catch
+        {
+        }
+
+        return finalFilePath;
+    }
+
+    /// <summary>
+    /// Creates an absolute file name for the provided <paramref name="filename"/> in the store.
+    /// </summary>
+    /// <param name="filename">The file name to use in the store.</param>
+    /// <returns>The absolute file name in the store.</returns>
+    public string GetFileName(string filename)
```

Comment:
What's this API used for? Why can't callers just use the BasePath to do whatever they need?